airbyte-workers/src/test/java/io/airbyte/workers/internal/EmptyAirbyteSourceTest.java

```diff
@@ -0,0 +1,263 @@
+        new StreamDescriptor().withName("b"),
+        new StreamDescriptor().withName("c")
+    );
+
+    List<io.airbyte.config.StreamDescriptor> streamToReset = Lists.newArrayList(
+        new io.airbyte.config.StreamDescriptor().withName("a"),
+        new io.airbyte.config.StreamDescriptor().withName("b"),
+        new io.airbyte.config.StreamDescriptor().withName("c")
+    );
+
+    ResetSourceConfiguration resetSourceConfiguration = new ResetSourceConfiguration()
+        .withStreamsToReset(streamToReset);
+    WorkerSourceConfig workerSourceConfig = new WorkerSourceConfig()
+        .withSourceConnectionConfiguration(Jsons.jsonNode(resetSourceConfiguration))
+        .withState(new State()
+            .withState(Jsons.jsonNode(createGlobalState(streamDescriptors, Jsons.emptyObject())))
+        );
+
+    emptyAirbyteSource.start(workerSourceConfig, null);
+
+    Optional<AirbyteMessage> maybeMessage = emptyAirbyteSource.attemptRead();
+    Assertions.assertThat(maybeMessage)
+        .isNotEmpty();
+
+    AirbyteMessage message = maybeMessage.get();
+    Assertions.assertThat(message.getType()).isEqualTo(Type.STATE);
+
+    /*
+      The comparison could be what it is bellow but it makes it hard to see what is the diff. It has been break dow into multiples assertions.
+      (same comment in the other tests)
+
+      AirbyteStateMessage expectedState = new AirbyteStateMessage()
+       .withStateType(AirbyteStateType.GLOBAL)
+       .withGlobal(
+         new AirbyteGlobalState()
+           .withSharedState(Jsons.emptyObject())
+           .withStreamStates(
+             Lists.newArrayList(
+               new AirbyteStreamState().withStreamState(null).withStreamDescriptor(new StreamDescriptor().withName("a")),
+               new AirbyteStreamState().withStreamState(null).withStreamDescriptor(new StreamDescriptor().withName("b")),
+               new AirbyteStreamState().withStreamState(Jsons.emptyObject()).withStreamDescriptor(new StreamDescriptor().withName("c"))
+             )
+           )
+      );
+
+      Assertions.assertThat(stateMessage).isEqualTo(expectedState);
+     */
+    AirbyteStateMessage stateMessage = message.getState();
+    Assertions.assertThat(stateMessage.getStateType()).isEqualTo(AirbyteStateType.GLOBAL);
+    Assertions.assertThat(stateMessage.getGlobal().getSharedState()).isNull();
+    Assertions.assertThat(stateMessage.getGlobal().getStreamStates())
+        .map(streamState -> streamState.getStreamDescriptor())
+        .containsExactlyElementsOf(streamDescriptors);
+    Assertions.assertThat(stateMessage.getGlobal().getStreamStates())
+        .map(streamState -> streamState.getStreamState())
+        .containsOnlyNulls();
+
+    Assertions.assertThat(emptyAirbyteSource.attemptRead())
+        .isEmpty();
+  }
+
+  @Test
+  public void testGlobalPartial() throws Exception {
+    final String NOT_RESET_STREAM_NAME = "c";
+
+    List<StreamDescriptor> streamDescriptors = Lists.newArrayList(
```

Comment:
can you make sure your intellij is configured to add `final` please ([link to docs](https://docs.airbyte.com/contributing-to-airbyte/code-style#configure-java-style-for-intellij))

Reply:
Done, save actions disabled it self sometimes